hastexo/hastexo.py

```diff
@@ -126,6 +133,17 @@ def parse_xml(cls, node, runtime, keys, id_generator):
                 text = textwrap.dedent(text)
 
                 block.tests.append(text)
+            elif child.tag == "provider":
+                name = child.attrib["name"]
+                if not name:
+                    raise KeyError("name")
+                capacity = child.attrib.get("capacity", None)
```

Comment:
Since this defaults to `None`, what is the defined behavior if the user sets `0`, or a negative number?